presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1348,11 +1356,27 @@ else if (expression instanceof LongLiteral) {
                     }
 
                     // otherwise, just use the expression as is
+                    boolean orderByOnExpression = false;
                     if (orderByExpression == null) {
                         orderByExpression = expression;
+                        orderByOnExpression = true;
                     }
 
                     ExpressionAnalysis expressionAnalysis = analyzeExpression(orderByExpression, sourceScope);
+                    if (orderByOnExpression) {
```

Comment:
Not sure if I can, because `ExpressionAnalysis expressionAnalysis` in the middle. I was thinking of `if (orderByExpression == expression)` instead of flag, but preferred safe.